### ISSUE TITLE
MSBuild error (code=1)

### ISSUE THREAD

**Issue by shllybkwrm** [Microsoft/MixedRealityToolkit-Unity#947]
I'm attempting to deploy my app from Unity to HL and getting this error with no explanation:

>`MSBuild error (code = 1)
UnityEngine.Debug:LogError(Object)
HoloToolkit.Unity.BuildDeployTools:BuildAppxFromSLN(String, String, Boolean, String, String, Boolean, Boolean) (at Assets/HoloToolkit/Build/Editor/BuildDeployTools.cs:253)
HoloToolkit.Unity.BuildDeployWindow:BuildAll(Boolean) (at Assets/HoloToolkit/Build/Editor/BuildDeployWindow.cs:650)
HoloToolkit.Unity.<OnGUI>c__AnonStorey0:<>m__0() (at Assets/HoloToolkit/Build/Editor/BuildDeployWindow.cs:181)
UnityEditor.EditorApplication:Internal_CallDelayFunctions()`

Any thoughts?  I just updated to the toolkit release for Unity 2017.1.1f1 because I thought it was supposed to fix MSBuild issues.

**Comment by shllybkwrm** [comment]
For further reference, I also attempted to build in VS17.  Here is the error there:

> `C:\Program Files (x86)\MSBuild\15.0\.Net\CoreRuntime\Microsoft.Net.CoreRuntime.targets(193,9): error : Framework resource extraction failed. This implementation is not part of the Windows Platform FIPS validated cryptographic algorithms.`

**Comment by StephenHodgson** [comment]
What's your build settings?

**Comment by shllybkwrm** [comment]
Sorry for the delay.  Here is what the HLTK build window set them as:
![image](https://user-images.githubusercontent.com/2496635/30282519-84c7ea42-96e3-11e7-8296-4c76937af01c.png)
I tried adjusting them as well but same error with these:
![image](https://user-images.githubusercontent.com/2496635/30282761-392560be-96e4-11e7-8bfd-91308053e742.png)

**Comment by StephenHodgson** [comment]
Try un checking Unity C# Projects and use the 14393 sdk.

**Comment by StephenHodgson** [comment]
Then open using Visual Studio and try building from there.

I think the issue you're running into might be related to your code base.  The toolkit doesn't rely on any cryptographic libraries.

**Comment by shllybkwrm** [comment]
Thanks, that allowed me to build successfully.  It seems like my VS17 install was missing part of the Windows 10 SDK that it didn't notify me about earlier.  I'll complete the install and report back if that fixes my issues.

**Comment by shllybkwrm** [comment]
Still no luck.  My code also isn't using cryptographic libraries, so I'll have to do some digging to see where this could be coming from.
If it's helpful, I'm running into this issue after rewriting my .NET networking code to instead do something similar to the CustomMessages.cs example code.  I can provide the code or snippets if anyone has some insight, but otherwise I guess I'll have to go through it myself.

`3>------ Build started: Project: ARInterface, Configuration: Release x86 ------`
`3>C:\Users\sbagchi\Documents\Vuforia\ARDemo\ARInterface-Unity\ARInterface\UWP\ARInterface\App.cs(56,13,56,71): warning CS0618: 'ApplicationView.SuppressSystemOverlays.set' is obsolete: 'Use the TryEnterFullScreen method and IsFullScreenMode property instead of SuppressSystemOverlays. For more info, see MSDN.'`
`3>  ARInterface -> C:\Users\sbagchi\Documents\Vuforia\ARDemo\ARInterface-Unity\ARInterface\UWP\ARInterface\bin\x86\Release\ARInterface.exe`
`3>C:\Program Files (x86)\MSBuild\15.0\.Net\CoreRuntime\Microsoft.Net.CoreRuntime.targets(193,9): error : Framework resource extraction failed. This implementation is not part of the Windows Platform FIPS validated cryptographic algorithms.`
`========== Build: 2 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========`

**Comment by shllybkwrm** [comment]
Progress!  I followed these steps to remove the cryptographic error and was able to build successfully.  I got this error the first time I attempted to deploy, but it seems to have gone away on successive tries?

![image](https://user-images.githubusercontent.com/2496635/30289795-47a9bbb4-96fb-11e7-85ab-b815eda1804f.png)

The app isn't working as expected (Unity UI elements aren't showing, Vuforia image recognition has some unexpected behavior, etc.), but that's a different issue, so I'm going to close this issue.  Thanks for the help @StephenHodgson!
